### PR TITLE
RUMM 2180 Add eslint rules

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

Bring the project closer to other datadog projects by adding some of the eslint rules in use

### Additional Notes

Please review https://github.com/DataDog/dd-sdk-reactnative/pull/177 first
- be careful with the conversion of double equality into triple equality

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [x] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

**Comment by 0xnm** [comment]
@louiszawadzki Let's maybe target branch of #177 to reduce the final diff and simplify the review? Once #177 is merged target here will be updated.

**Review by mariusc83 — APPROVED** [review]

File: .eslintrc.js [hunk 1]
```diff
@@ -17,11 +17,314 @@ module.exports = {
...
+        'react-hooks',
+        '@typescript-eslint',
+        'arca'
+    ],
     rules: {
         'react/prop-types': 0,
         '@typescript-eslint/no-inferrable-types': 0,
         '@typescript-eslint/no-explicit-any': 0,
-        'prettier/prettier': 'error'
+        'prettier/prettier': 'error',
```
Comment (0xnm) [inline .eslintrc.js:32]: IMO we should add the comment saying from where this rule set is coming from, to keep it in sync with upstream rule set changes.

File: package.json [hunk 2]
```diff
@@ -51,6 +54,10 @@
     "typescript": "4.1.3",
     "genversion": "3.0.2"
   },
+  "resolutions": {
+    "**/@types/react": "^16.13.1",
+    "**/@types/react-native": "0.63.42"
```
Comment (0xnm) [inline package.json:59]: Is this a correct version? I can see only version `0.63.4` for RN, but not `0.63.42` (but maybe types package has a different versioning).

File: packages/core/src/rum/instrumentation/DdRumUserInteractionTracking.tsx [hunk 3]
```diff
@@ -65,7 +65,7 @@ export class DdRumUserInteractionTracking {
         ) => {
             return originalMemo(component, (prev, next) => {
                 if (!next.onPress || !prev.onPress) {
-                    return !!propsAreEqual
```
Comment (0xnm) [inline packages/core/src/rum/instrumentation/DdRumUserInteractionTracking.tsx:68]: What is the motivation behind this change? Why we used `!!` (well, probably it is a conversion from `null|undefined` to boolean) and now it is not needed?

**Review by 0xnm — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (louiszawadzki) [reply package.json:59]: yes this is the version for the types package

File: packages/core/src/rum/instrumentation/DdRumUserInteractionTracking.tsx (hunk 3 above)
Comment (louiszawadzki) [reply packages/core/src/rum/instrumentation/DdRumUserInteractionTracking.tsx:68]: Because we chain this to a ternary operator (`?`), so using `!!` actually does not make any difference here and can be removed. I believe I should not have added it in the first place.

**Review by 0xnm — APPROVED** [review]
